### ISSUE TITLE
broken feature: resubmitting simulations with the ENSTDIF-workaround after model crash due to instability

### ISSUE THREAD

**Issue by christian-stepanek** [esm-tools/esm_tools#58]
**Description of the problem:**
Sometimes it is necessary to resubmit a simulation after it crashed due to numeric instability. The traditional method is to slightly increase energy dissipation in the model by increasing the value of the namelist parameter ENSTDIF by 1/10000 for one year only. Normally, this will help the model over the instability and from the following year on the model simulation can execute unperturbed.

In the here employed newest release of the esm-tools it appears that there is no possibility (see description below) to modify the ENSTDIF parameter for one year only, hence leading to a dead end for simulations where numeric instability occurs (or, where the automatic instability detection wrongly assumes instability to occur, see issue #57).

Assume the simulation crashes in model year 2013 due to either model instability or erroneously detected model instability. According to the suggested method the experiment yaml should be modified by adding one line to the echam section that disturbs echam during the problematic year:
```
        disturbance_years: [ 2013 ]
```
After this is done, according to traditional experience with the esm-tools, the simulation should be resubmitted employing the esm_runscripts command line switch -U in order to update the run script at the experiment run location:
```
stepanek@ollie0:~/esm-tools_v4/esm_tools/myrunscripts$ esm_runscripts myinitialtest_yearly.yaml -e myinitialtest_yearly -U
```
Sadly, this does not lead to the expected update of the runscript or the namelist in the work folder:
```
stepanek@ollie0:~/esm-tools_v4/esm_tools/myrunscripts$ diff /work/ollie/stepanek/esm-experiments_v4/myinitialtest_yearly/run_20130101-20131231/scripts/myinitialtest_yearly.yaml myinitialtest_yearly.yaml 
18a19
>         disturbance_years: [ 2013 ]
stepanek@ollie0:~/esm-tools_v4/esm_tools/myrunscripts$ cat /work/ollie/stepanek/esm-experiments_v4/myinitialtest_yearly/run_20130101-20131231/work/namelist.echam | grep -i ENSTDIF
stepanek@ollie0:~/esm-tools_v4/esm_tools/myrunscripts$ cat /work/ollie/stepanek/esm-experiments_v4/myinitialtest_yearly/run_20130101-20131231/work/namelist.echam | grep -i dynctl

```
In summary: The run script does not get updated as would be expected from the use of the -U switch, and, unsurprisingly, the namelist settings do not get updated as well.

**employed run script (on Ollie):**
/home/ollie/stepanek/esm-tools_v4/esm_tools/myrunscripts/myinitialtest_yearly.yaml

**employed esm_tools versions:**
esm_archiving : unknown version!
esm_autotests : unknown version!
esm_calendar : 4.0.1
esm_database : 4.0.0
esm_environment : 4.0.1
esm_master : 4.0.2
esm_parser : 4.0.2
esm_profile : 4.0.0
esm_rcfile : 4.0.0
esm_runscripts : 4.0.3
esm_tools : 4.0.9
esm_plugin_manager : 4.0.1
esm_version_checker : 4.0.2

**Comment by pgierz** [comment]
Hi Christian,

this one is next on my list, because we definitely will need it. Could you please give me an opinion on the following:\

A) The job should resubmit itself (with a counter of how often running, and try resubmitting a max number of times before it gives up)

B) The program should just stop. This normally *would* happen anyway, but it runs into the wall clock barrier at some point.

C) Do nothing, and let the wall clock take over.

In any case, regardless of the choice above, we need a mechanism to selectively patch the windspeed fix for specific runs. This is already implemented [here](https://github.com/esm-tools/esm_runscripts/blob/568cb57686d8a365e109b6698d70d5b0aa6d1851/esm_runscripts/namelists.py#L63), but still needs a test/sanity check before I merge it.

I also will likely move this issue to the esm-runscripts repository to keep tabs on what is happening where...so don't be surprised if it vanishes ;-)

**Comment by pgierz** [comment]
Two things were wrong here:

* `-U` did not work (that has been fixed)
* wind speed updates were not applied correctly: this has also been fixed, but it does not happen automatically. 

If there is a desire for an automatic fix and resubmission, please open up a new issue. I am going through the old ones and cleaning up before we start release of version 6.
